website/content/docs/agent-and-proxy/agent/generate-config/index.mdx

```diff
@@ -0,0 +1,110 @@
+specified secret. If the secret `-path` ends with `/*`, the command will
+attempt to recurse through the secrets tree rooted at the given path,
+generating `env_template` entries for each encountered secret. Currently,
+only [kv-v1](/vault/docs/secrets/kv/kv-v1) and
+[kv-v2](/vault/docs/secrets/kv/kv-v2) paths are supported.
+
+The command specified in the `-exec` option will be used to generate an
+`exec` entry, which will tell Vault Agent which child process to run.
+
+In addition to the `env_template` entries, the command generates an `auto_auth`
+section with `token_file` authentication method. While this method is very
+convenient for local testing, it should **NOT** be used in production. Please
+refer to the [Auto-Auth docs](/vault/docs/agent-and-proxy/autoauth) for
+a list of production-ready `auto_auth` methods that you can use instead.
+
+By default, the file will be generated in the local directory as `agent.hcl`
+unless a path is specified as an argument.
+
+## Examples
+
+Before generating a configuration file, let's insert a secret `foo`:
+
+```shell-session
+$ vault kv put -mount=secret foo user="admin" password="s3cr3t"
+```
+
+Generate an agent configuration file which will reference `secret/foo`:
+
+```shell-session
+$ vault agent generate-config \
+                -type="env-template" \
+                -exec="./my-app arg1 arg2" \
+                -path="secret/foo" \
+                    my-config.hcl
+
+Successfully generated "my-config.hcl" configuration file!
+Warning: the generated file uses 'token_file' authentication method, which is not suitable for production environments.
+```
+
+This will produce `my-config.hcl` file in the current directory with contents
+similar to the following:
+
+```hcl
+auto_auth {
+
+  method {
+    type = "token_file"
+
+    config {
+      token_file_path = "/Users/avean/.vault-token"
+    }
+  }
+}
+
+template_config {
+  static_secret_render_interval = "5m"
+  exit_on_retry_failure         = true
+}
+
+vault {
+  address = "http://localhost:8200"
+}
+
+env_template "FOO_PASSWORD" {
+  contents             = "{{ with secret \"secret/data/foo\" }}{{ .Data.data.password }}{{ end }}"
+  error_on_missing_key = true
+}
+env_template "FOO_USER" {
+  contents             = "{{ with secret \"secret/data/foo\" }}{{ .Data.data.user }}{{ end }}"
+  error_on_missing_key = true
+}
+
+exec {
+  command                   = ["./my-app", "arg1", "arg2"]
+  restart_on_secret_changes = "always"
+  restart_stop_signal       = "SIGTERM"
+}
+```
+
+## Usage
+
+The following flags are available in addition to the [standard set of
+flags](/vault/docs/commands) included in all commands.
+
+- `type` `(string: <required>)` - The type of configuration file to generate;
+  currently, only 'env-template' is supported.
+
+- `path` `(string: "")` - Path to a kv-v1 or kv-v2 secret
+  (e.g. secret/data/foo, kv-v2/prefix/*); multiple secrets and tail '*'
```

Comment:
I think you'll need to backslash the asterisks, as they're making italics, right now:

<img width="656" alt="image" src="https://github.com/hashicorp/vault/assets/1594272/2651d7fc-13a1-4740-be25-48d7848f4112">